python/cudf/dataframe/dataframe.py

```diff
@@ -515,28 +515,28 @@ def op(lhs, rhs):
             for k, col in enumerate(self._cols):
                 result[col] = getattr(self._cols[col], fn)(other[k])
         elif isinstance(other, DataFrame):
-            max_num_rows = max(self.shape[0], other.shape[0])
+
+            lhs, rhs = _align_indices(self, other)
```

Comment:
We shouldn't align these if they're already equal since that would be expensive, and potentially cause incorrect behavior.

Reply:
@kkraus14 that's the first check in the `_align_indices()` function :-)

Reply:
Woops, correct!